### DISCUSSION TITLE
[Feature][mobile] Multiselection save/download

### DISCUSSION THREAD

**Discussion by Dschogo** [immich-app/immich#5068] [Feature Request]
### The feature

Add an option to download/save/make offline available when multiple images are selected. Basically inverse upload button.

The only current way to make images offline available is: 
Viewing each image and pressing manually download/save. (or downloading via web view)

With that feature the procedure would be:
- multi-select images which are "online only"
- select the new save/download
- app downloads these images to the gallery folder (called pictures like from the individual image download currently) [android]

Wouldn't make that option available for shared albums, since that folder ("pictures") could be included for backup and create cross user duplicates. For that maybe an "make album offline available?" and create a new folder for it? (out of scope for this one)

Thx to all immich devs :heart:

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by aisbergde** [comment]
I would not distinguish between shared and not shared albums, it could be in both of them.

Avoiding to download duplicates should be independent from the way, how a download is initiated (by single selecting, multi selecting, time range, whole album or whatever feature request could be found here). If avoiding duplicate upload is already done with checksum, this would be possible for downloading too.

**Comment by hofq** [comment]
I stumbled on this today. Would save a lot of time!

**Comment by nachadan** [comment]
it would be great to have this option. When i upload multiple photos through PC then i have to download every photo each by each to have it offline on device. Thanks

**Comment by lukethehawk** [comment]
it would be great to have a select-all button, to select all the photos and do this download in 2 taps

**Comment by DrieWielRr** [comment]
+1, this is a must for sharing pictures on other apps like instagram for example, share with app does not allow you to create a regular post only a reel or story, so content needs to be selected manually within the insta-app to create a regular post, we share immich-albums and having too download one-by-one from a shared album is getting tedious..
Imho you are yourself responsible for having cross-user duplicates because you also backup the folder where the images are downloaded to, a solution would be able to select the storage location on the phone when tapping download.

**Comment by catacgc** [comment]
Another cool feature will be to be able to download from multiple selections from search results. My use case is a simple one , I am trying to search for landscape photos to use as wallpapers on my phone . The search works great , but I cannot download the photos

**Comment by ryanrpj** [comment]
+1 for this

**Comment by NicholasFlamy** [comment]
> For that maybe an "make album offline available?" and create a new folder for it? (out of scope for this one)

This specific idea mentioned in the post is related to #7627 in case anyone's wondering.

**Comment by TiagoDM** [comment]
I was about to create a feature request ticket, but stumbled upon this one and i think its quite similar to the feature i wanted in immich android app. I think it would be a must to be able to download whole albums to the device, since i think it's not possible to do that at this point. From the web interface it is possible to download albums, but not in android app. Please do add that feature. Cheers.

**Reply by NicholasFlamy** [reply]
No, that's #7627

**Comment by Ainges** [comment]
I think there's a workaround – at least on iOS: you can select multiple photos, tap the Share button, and then choose "Save [x] Images" from the system dialog. In my testing, the Immich app recognizes these photos after restarting the app.

Does anyone know if there are any downsides to this approach? A native solution would definitely be better, but for now, I'm quite happy with this workaround.

**Reply by vladd49** [reply]
It doesn't work, photos appear duplicated when I download them this way and restart the app.

The lack of such a basic feature (bulk download) is a major disappointment in Immich which in all other respects is great. I often need to download lots of photos to make them available in iCloud, but this is always a pain. The only way to do this is to download one-by-one, but Immich app rescans the whole library after I download every 1-2 photos. This makes my phone hot like an iron, very unresponsive and the whole process takes 1-2 hours.

Whish the developers would pay attention to this. It's a big fly in the ointment.

**Comment by nachadan** [comment]
When i need to download in bulk i open it through web browser (local ip), hit ctrl+a and then select download all. Then i connect phone to a computer and transfer it to the new phone. It just rescan gallery. But I would love to see this function in the future, as I was one of the first who recommended this.


On May 31, 2025 1:18:37 PM GMT+02:00, vladd49 ***@***.***> wrote:
>It doesn't work, photos appear duplicated when I download them this way and restart the app.
>
>The lack of such a basic feature (bulk download) is a major disappointment in Immich which in all other respects is great. I often need to download lots of photos to make them available in iCloud, but this is always a pain. The only way to do this is to download one-by-one, but Immich app rescans the whole library after I download every 1-2 photos. This makes my phone hot like an iron, very unresponsive and the whole process takes 1-2 hours.
>
>Whish the developers would pay attention to this. It's a big fly in the ointment.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/immich-app/immich/discussions/5068#discussioncomment-13328673
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
DriveMont

**Comment by JobiJoba** [comment]
PR #18878 has been submitted (and approved) for this feature

**Reply by AlCalzone** [reply]
Maybe I'm blind, but how do I actually use this in the iOS app?
